tests/integration/test_integration.py

```diff
@@ -192,6 +192,22 @@ def test_payload_too_large(encoding, monkeypatch):
         log.error.assert_not_called()
 
 
+@pytest.mark.skipif(AGENT_VERSION == "testagent", reason="FIXME: Test agent doesn't support this for some reason.")
+def test_resource_name_too_large(monkeypatch):
+    SIZE = 1 << 12  # 4KB
+    monkeypatch.setenv("DD_TRACE_API_VERSION", "v0.5")
+    monkeypatch.setenv("DD_TRACE_WRITER_BUFFER_SIZE_BYTES", str(SIZE))
+
+    t = Tracer()
+    assert t._writer._buffer_size == SIZE
+    s = t.trace("operation", service="foo")
+    s.resource = "B" * (SIZE + 1)
+    try:
+        s.finish()
+    except ValueError:
```

Comment:
we don't get a `ValueError` here right? since we now truncate the string, it should fit. So the `s.finish()` should succeed.

do we need to validate that the resulting payload replaces the resource name with the truncated string?

Reply:
This test validates that large strings don't cause `ValueError`, which they do on current 1.x.

Reply:
do we need the `try/except`? since if it did raise the error then the test would also just fail.

we could get the same from:

```python
# does not raise a ValueError
s.finish()
```

Reply:
do you have a preference/opinion the approach (or I suppose you do since you wrote with the try/except, just curious your thoughts on that approach)?

Reply:
Using a `try/except` makes it explicit that the test checks for regressions on a call that used to throw an error, whereas letting the test fail by allowing the exception to bubble up keeps that information implicit. This is a small attempt on my part to follow the Zen of Python maxim "explicit is better than implicit".